contracts/datastore/DataStore.sol

```diff
@@ -0,0 +1,362 @@
+pragma solidity ^0.5.0;
+
+import "../interfaces/ISecurityToken.sol";
+import "../interfaces/IOwnable.sol";
+import "../interfaces/IDataStore.sol";
+import "./DataStoreStorage.sol";
+
+/**
+ * @title Data store contract that stores data for all the modules in a central contract.
+ */
+contract DataStore is DataStoreStorage, IDataStore {
+    //NB To modify a specific element of an array, First push a new element to the array and then delete the old element.
+    //Whenver an element is deleted from an Array, last element of that array is moved to the index of deleted element.
+    //Delegate with MANAGEDATA permission can modify data.
+
+    modifier onlyAuthorized() {
+        bool isOwner = msg.sender == IOwnable(address(securityToken)).owner();
+        require(isOwner || 
+            securityToken.isModule(msg.sender, DATA_KEY) || 
+            securityToken.checkPermission(msg.sender, address(this), MANAGEDATA), 
+            "Unauthorized"
+        );
+        _;
+    }
+
+    modifier validKey(bytes32 _key) {
+        require(_key != bytes32(0), "Missing key");
+        _;
+    }
+
+    modifier validInputLength(uint256 _keyLength, uint256 _dataLength) {
```

Comment:
I actually had it named `validArrayLength` earlier but changed it to `validInputLength` because it seemed better to me.
Anyway, I don't really have a strong preference for any. I'll change it back to `validArrayLength`.